### PR TITLE
Dummy PR [Do Not Merge Yet]

### PR DESCRIPTION
**Pull request by dannyallover**

### CODE REVIEW

**Comment by dannyallover** [comment]
PTAL @aronsar

File: domrl/agents/provincial_agent.py [hunk 1]
```diff
@@ -77,6 +82,32 @@ def hand_contains(state_view, card_name):
...
+    estate_count = state_view.supply_piles["Estate"].qty
+
+    if coins >= 8 and prov_index != [0]:
+        return prov_index
+    if coins >= 5 and duchy_index != [0] and prov_count < duchy_buy_threshold:
+        return duchy_index
+    if coins >= 1 and estate_index != [0] and prov_count < estate_buy_threshold:
+        return estate_index
+
+    for i in range(len(buy_menu)):
```
Comment (aronsar) [inline domrl/agents/provincial_agent.py:114]: for i, (card, cost, amount_to_buy) in enumerate(buy_menu):

File: domrl/engine/state_view.py [hunk 2]
```diff
@@ -24,6 +24,20 @@ def __init__(self, player, is_player: bool):
         self.all_cards = [card for card in player.discard_pile] + \
             [card for card in player.hand] + \
             [card for card in player.draw_pile]
+        self.all_card_names = [card.name for card in player.discard_pile] + \
+            [card.name for card in player.hand] + \
+            [card.name for card in player.draw_pile]
+        def CountFrequency(my_list):
```
Comment (aronsar) [inline domrl/engine/state_view.py:32]: from collections import Counter
Counter(list) returns a dictionary that holds the same info as freq on line 37

File: domrl/agents/big_money_agent.py [hunk 3]
```diff
@@ -18,10 +18,31 @@ def find_card_in_decision(decision, card_name):
                 return [idx]
     return [0]
 
+# helper to sort a list of tuples by the second element
+def Sort_List_Of_Tuples(tup):
```
Comment (benzyx) [inline domrl/agents/big_money_agent.py:22]: I don't think this needs to be a special function. You can just do:

`tup.sort(key = lambda x: x[1]) 
`

File: domrl/agents/big_money_agent.py [hunk 4]
```diff
@@ -18,10 +18,31 @@ def find_card_in_decision(decision, card_name):
...
+                tup[j]= tup[j + 1]
+                tup[j + 1]= temp
+    return tup
 
 def get_minimum_coin_card(decision):
-    card_coins = [c.coins for c in decision.moves.card]
-    return [np.argmin(card_coins)]
+    priority = []
+    for i in range(len(decision.moves)):
+        priority.append((decision.moves[i].card.cost, i))
```
Comment (benzyx) [inline domrl/agents/big_money_agent.py:36]: The more pythonic approach is

`priority = [(move.card.cost, i) for i, move in enumerate(decision.moves)]`

You can further sort this in the same line:

`priority = [(move.card.cost, i) for i, move in enumerate(decision.moves)].sort(key=lambda x: x[1])`

File: domrl/agents/provincial_agent.py [hunk 5]
```diff
@@ -169,36 +206,38 @@ def provincial_reaction_bureaucrat(decision):
...
-    for i in range(1, len(decision.moves)):
+    for i in range(len(decision.moves)):
         move = decision.moves[i]
         if hasattr(move, 'card') and move.card.name in discard_priority:
             cards_ordered.append((i, discard_priority[move.card.name]))
         elif hasattr(move, 'card'):
             cards_ordered.append((i, -100 - move.card.cost)) # ranking provincial uses
 
-    return [Sort_List_Of_Tuples(cards_ordered)[0][0]]
+    cards_ordered = Sort_List_Of_Tuples(cards_ordered)
```
Comment (benzyx) [inline domrl/agents/provincial_agent.py:201]: don't use `Sort_List_Of_Tuples` here.

File: domrl/engine/state_view.py [hunk 6]
```diff
@@ -24,6 +24,20 @@ def __init__(self, player, is_player: bool):
         self.all_cards = [card for card in player.discard_pile] + \
             [card for card in player.hand] + \
             [card for card in player.draw_pile]
+        self.all_card_names = [card.name for card in player.discard_pile] + \
```
Comment (benzyx) [inline domrl/engine/state_view.py:28]: I'm not a fan of setting this. I think the agent can define their own function to get these.

File: domrl/engine/state_view.py (hunk 2 above)
Comment (benzyx) [reply domrl/engine/state_view.py:32]: i agree

File: provincial_vs_provincial.py [hunk 7]
```diff
@@ -0,0 +1,58 @@
+from domrl.engine.game import Game
```
Comment (benzyx) [inline provincial_vs_provincial.py:1]: could you move this into examples/

**Review by benzyx — CHANGES_REQUESTED** [review]
quick read through.

**Comment by dannyallover** [comment]
[CLOSING] Since I am submitting changes directly to [domray](https://github.com/aronsar/domray), which has DomRL embedded in it.

**Pull request closed without merging**